### PR TITLE
GitHub PR templates and Pre-commit file

### PR DESCRIPTION
**Pull request by vara-bonthu**
*Issue #, if available:*

- Adds PR and Issue templates
- Adds pre-commit file 

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,29 @@
+
```
Comment (nabuskey) [inline .github/PULL_REQUEST_TEMPLATE.md:1]: We should remind people to break up massive PRs into smaller chunks whenever possible. Definitely do not want to review massive PRs. 
So we could say something like `We prefer small, well tested pull requests. Please ensure your pull requests are self-contained, and commits are squashed`

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+### More
+
+- [ ] Yes, I have tested the PR using my local account setup  (Provide any test evidence report under Additional Notes)
+
+- [ ] Yes, I have added a new example under [examples](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/examples) to support my PR
+
+- [ ] Yes, I have updated the [docs](https://github.com/aws-samples/crossplane-aws-blueprints/tree/main/doc) for this feature
+
```
Comment (nabuskey) [inline .github/PULL_REQUEST_TEMPLATE.md:19]: Add something about linking to issues when applicable. 
```
Yes, I have linked to an issue or feature request (applicable to PRs that solves a bug or a feature request)
```

File: examples/aws-provider/composite-resources/dynamodb/dynamodb-on-demand-partition.yaml [hunk 3]
```diff
@@ -26,4 +26,3 @@ spec:
   tableIndex:
     hashKeyName: hashKey
     hashKeyType: S
-
```
Comment (nabuskey) [inline examples/aws-provider/composite-resources/dynamodb/dynamodb-on-demand-partition.yaml:29]: Why are newlines at the end being removed? We should keep them.

**Review by nabuskey — APPROVED** [review]
Add suggestions.

File: examples/aws-provider/composite-resources/dynamodb/dynamodb-on-demand-partition.yaml (hunk 3 above)
Comment (vara-bonthu) [reply examples/aws-provider/composite-resources/dynamodb/dynamodb-on-demand-partition.yaml:29]: Its deleting extra lines which are more than one in each file. I can see the empty line in the code but not sure why its displaying like this in the PR.

We are using  `pre-commit-hooks` with `end-of-file-fixer` to remove unnecessary extra lines and keeps  only one new line

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 1 above)
Comment (vara-bonthu) [reply .github/PULL_REQUEST_TEMPLATE.md:1]: updated

File: .github/PULL_REQUEST_TEMPLATE.md (hunk 2 above)
Comment (vara-bonthu) [reply .github/PULL_REQUEST_TEMPLATE.md:19]: updated

File: examples/aws-provider/composite-resources/dynamodb/dynamodb-on-demand-partition.yaml (hunk 3 above)
Comment (nabuskey) [reply examples/aws-provider/composite-resources/dynamodb/dynamodb-on-demand-partition.yaml:29]: It deleted the last newlines in files that were supposed to be there. See https://raw.githubusercontent.com/aws-samples/crossplane-aws-blueprints/main/examples/aws-provider/composite-resources/dynamodb/dynamodb-provisioned-composite-gsi.yaml

**Pull request merged**